### PR TITLE
Store Form Data

### PR DESCRIPTION
**Pull request by mohametalmeari**
- [x] Store form data in local storage.
- [x] Fill the form with the stored data when page is loaded.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -186,3 +186,47 @@ form.addEventListener('submit', (event) => {
...
+}
+
+if (storageAvailable('localStorage')) {
+  if (localStorage.getItem('name')) {
+    fillForm();
+  }
+
+  form.addEventListener('submit', () => {
+    populateStorage();
+  });
```
Comment (codepantha) [inline script.js]: - [ ] According to requirements, after filling the form, and not submitting it, the entered values should be saved in the browser. But in your case, I noticed that you only save to `localStorage` when the submit button is clicked. Please rewrite this functionality to fulfill the requirement 🙏

File: script.js [hunk 2]
```diff
@@ -186,3 +186,47 @@ form.addEventListener('submit', (event) => {
...
+      && storage.length !== 0
+    );
+  }
+}
+
+function populateStorage() {
+  localStorage.setItem('name', document.getElementById('name').value);
+  localStorage.setItem('email', document.getElementById('email').value);
+  localStorage.setItem('message', document.getElementById('message').value);
+}
```
Comment (codepantha) [inline script.js]: - [ ] I noticed that you have separate entries for each of the form inputs (name, email, and message).
Please ensure that the form data for all inputs are stored in a single object, saved to a single local storage registry as per requirements 👍 

image

### HINT: 👇 
``` js
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};
```
Then turn it into a string before saving to local storage :
```js
localStorage.setItem('contactFormInput', JSON.stringify(formData));
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Save to `localStorage` 📌 
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a
- 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by aamirkhan2478 — APPROVED** [review]
Hi @mohametalmeari and @OTMANE-ECHCHAFYKY , 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly
✅ No linter error
✅ Good commit messages
✅ Good descriptive PR
✅ Well done you did all requirement fulfil mentioned by the previous reviewer :fireworks: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**